app/controllers/users_controller.rb

```diff
@@ -1,20 +1,36 @@
+# frozen_string_literal: true
+
 class UsersController < ApplicationController
+  before_action :set_user, only: %i[edit update]
+
   def new
-    @r = User.new
+    @user = User.new
   end
 
   def create
     @r = User.create(user_params)
 
-   if @r
-    redirect_to new_user_path
-   else
-    render :new
-   end
+    if @r
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def update
+    @user.update(user_params)
   end
 
   private
+
   def user_params
+    # params.permit(:username, :email, :password)
     params.require(:user).permit(:username, :email, :password)
   end
+
+  def edit; end
```

Comment:
- [ ] this method, edit, should be filled

Reply:
The method has been removed, it is not necessary to edit form to work